addons/pkg/util/clusterbootstrapclone/clusterbootstrapclone.go

```diff
@@ -124,37 +127,72 @@ func (h *Helper) AddMissingSpecFieldsFromTemplate(clusterBootstrapTemplate *runt
+func addMissingFields(copyFrom, destination, keysToSkip map[string]interface{}) error {
 	for keyInFrom, valueInFrom := range copyFrom {
+		if keysToSkip != nil {
+			// There might be some keys we do not want to add to destination.
+			if _, exist := keysToSkip[keyInFrom]; exist {
+				continue
+			}
+		}
 		valueInTarget, exist := destination[keyInFrom]
 		if !exist || valueInTarget == nil {
-			// If keyInFrom does not exist in destination or valueInTo is nil, we need to copy valueInFrom and add to
+			// If keyInFrom does not exist in destination or valueInTarget is nil, we need to copy valueInFrom and add to
 			// destination.
 			if valueInFrom != nil {
 				valueInFromType := reflect.TypeOf(valueInFrom)
 				if valueInFromType.Kind() == reflect.Map {
+					// If the valueInFrom is a map, we copy it directly.
 					copiedVal, _, copyErr := unstructured.NestedFieldCopy(valueInFrom.(map[string]interface{}))
 					if copyErr != nil {
 						return copyErr
 					}
+					// unstructured.NestedFieldCopy() copies all fields, but we need to delete some fields which are meant
+					// to be skipped.
+					if removeErr := removeFields(copiedVal.(map[string]interface{}), keysToSkip); removeErr != nil {
+						return removeErr
+					}
 					destination[keyInFrom] = copiedVal
+				} else if valueInFromType.Kind() == reflect.Slice {
+					// If the valueInFrom is a slice, e.g., additionalPackages from a ClusterBootstrapTemplate, we iterate
+					// on each item of the slice and make a copy of it.
+					var copiedSlice []interface{}
+					for _, sliceItemInFrom := range valueInFrom.([]interface{}) {
+						if reflect.TypeOf(sliceItemInFrom).Kind() == reflect.Map {
+							// Instantiate an empty map and invoke addMissingFields() to add missing fields into it. Logically,
+							// this is doing a copy of the slice item and remove the keys which are meant to be skipped.
+							copiedMap := make(map[string]interface{})
+							if copyErr := addMissingFields(sliceItemInFrom.(map[string]interface{}), copiedMap, keysToSkip); copyErr != nil {
+								return copyErr
+							}
+							copiedSlice = append(copiedSlice, copiedMap)
+						} else {
+							// TODO: The following line is doing a shallow copy, we might want to revisit in the future.
+							// https://github.com/vmware-tanzu/tanzu-framework/issues/2785
+							copiedSlice = append(copiedSlice, sliceItemInFrom)
+						}
+					}
+					destination[keyInFrom] = copiedSlice
 				} else {
 					// TODO: Handle the primitive pointer, e.g., *int. Ideally we need to make a copy of the pointer
 					// instead of reassigning directly. It is a shallow copy with current approach, copyFrom and destination
 					// share the same underlying data. We are good at the moment because this function is internally
 					// used for handling ClusterBootstrapTemplateSpec which does not have any primitive pointers in its
 					// API definition.
+					// https://github.com/vmware-tanzu/tanzu-framework/issues/2785
```

Comment:
Since we don't support this in our data model, I would suggest log a warning(if not error) and don't do anything.